dom/nodes/pre-insertion-validation-hierarchy.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Validations where `child` argument is irrelevant.
+ * @param {Function} methodName
+ */
+function preInsertionValidateHierarchy(methodName) {
+  function insert(parent, node) {
+    if (parent[methodName].length > 1) {
+      parent[methodName](node, null);
+    } else {
+      parent[methodName](node);
+    }
```

Comment:
Is this stuff needed?

Reply:
Yes as `insertBefore` needs a second argument. I can't blindly pass `null` as it causes moving `node` into a `DocumentFragment`.